specs/dialog/dialog.md

```diff
@@ -0,0 +1,113 @@
+>
+> Like non-modal dialogs, modal dialogs contain their tab sequence. That is, Tab and Shift + Tab do not move focus outside the dialog. However, unlike most non-modal dialogs, modal dialogs do not provide means for moving keyboard focus outside the dialog window without closing the dialog.
+
+### Use Cases
+
+- Joanna has some friends coming over and is searching online for some adult beverage ideas. She visits a website of a local distillery and a form overlays the screen asking her to confirm she is over 21 years of age by entering her birthday.
+
+- Harold is in the process of removing old data sets from his company's content management system. He selects a group of data and clicks delete. A popup appears asking him to confirm that he wants to delete the selected files permanently.
+
+- Meredith is browsing news articles online and clicks a link to a local newspaper story about an upcoming heat wave in the region. She is informed that she has read through an allotted number of free articles for the month and must either login or become a subscriber of the news site to continue. Meredith tries to continue interacting with the web page, but she's unable to. Meredith decides to subscribe to the online newspaper because they are running a special.
+
+### Features
+
+- **Soft dismiss (light dismiss):** In certain implementations, app authors want a dialog to close when a user clicks outside the content region or presses the `esc` key with their keyboard.
+
+- **Focus management:** In all circumstances, focus should move to an element contained in the dialog. Per the W3C, "Unless a condition where doing otherwise is advisable, focus is initially set on the *first* focusable element" (emphasis mine). If an element within the dialog is set to be focused by an app author, our control should not override that behavior. In both modal and non-modal dialogs, tab focus should remain within the dialog content region and not move outside it. When a dialog the dialog is closed, users may also want a way to return focus to the element which invoked the dialog.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/dialog)
+- [Material UI](https://material-ui.com/components/dialogs/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/modals/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/modal/code/)
+- [Ant Design](https://ant.design/components/modal/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/modal-dialog)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/dialog)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/dialogs)
+
+---
+
+## Design
+
+### API
+
+*Component name:*
+- `fast-dialog`
+
+*Attributes:*
+- `modal` - boolean value for modal functionality. Sets aria-modal to true.
+- `visible` - boolean value to toggle the visibility of the dialog
```

Comment:
[hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) is a built-in attribute that I believe has the same function here. It might be worth considering adopting a more standard attribute for this purpose.

Reply:
I think we could adopt this - would the hidden selector work here if I wanted to style via CSS? In the event that a user changes the display property where the hidden attribute is, that will override the hidden behavior. In this case, I would want to be sure that the following would work:

HTML:
```html
<fast-dialog hidden>Foo bar</fast-dialog>
```
CSS:
```css
fast-dialog[hidden] {
    display: none;
}
```

Reply:
In my understanding, that should work. We'll provide a default host style in the dialog css as follows:

```css
:host([hidden]) { 
  display: none;
}
```

The consumer of the dialog can always provide their own css to override the host styles.